src/components/colorscale/calc.js

```diff
@@ -20,7 +20,7 @@ module.exports = function calc(trace, vals, containerStr, cLetter) {
 
     if(containerStr) {
         container = Lib.nestedProperty(trace, containerStr).get();
-        inputContainer = Lib.nestedProperty(trace._input, containerStr).get();
+        inputContainer = Lib.nestedProperty(trace, containerStr).get();
```

Comment:
@etpinard I got around to doing some manual restyling test for this. 

1. Initially the plot has specific `cmin`/`cmax` values; `cauto` false
2. I do `Plotly.restyle(gd, 'marker.cauto', true)`
3. Result is, the color palette now expands to the entire range of the data as it should
4. I do `Plotly.restyle(gd, 'marker.cauto', false)`
5. Result is, the colors on the markers get restored to what they were

Can you please give me a use case that would necessitate that the automatically calculated `cmin`/`cmax` values need to be persisted on `containerIn` for the `restyle` to work out?

It's not quite working yet for `line`, because `plot_api.js:162` has the `recalc` variable as `true` for the marker but `false` for the line, I'm looking into this.